### ISSUE TITLE
Blazor: Use of `--no-build` with `dotnet publish` breaks content with .NET 9 SDK

### ISSUE THREAD

**Issue by craigktreasure** [dotnet/aspnetcore#58321]
### Is there an existing issue for this?

- [x] I have searched the existing issues

## TLDR

* Change in behavior for `dotnet publish --no-build` when using the .NET 9 SDK, which causes incorrectly named files in the `wwwroot` folder and breaks content.
* When the `jsmodules.publish.manifest.json` resources are not named properly, there is no error indication as to what the problem is.

## Explanation

Upon upgrading to use the .NET 9 SDK, my production endpoint began showing as blank. No errors in the console and no 404 errors retrieving resources. Inspecting the page, the `body` tag has a `hidden-body` class causing it to be...well...hidden. The `loading-theme` component also has `mode="null"`, which isn't expected:
```xml
<loading-theme storage-name="theme">
  <fluent-design-theme mode="null" primary-color="Default"></fluent-design-theme>
</loading-theme>
```
If I remove the `hidden-body` class exposing the page, the theme is not correct. If I attempt to interact with the theme components, I start to see errors in the console:

```text
Error: Microsoft.JSInterop.JSException: Cannot read properties of undefined (reading 'clearLocalStorage')
TypeError: Cannot read properties of undefined (reading 'clearLocalStorage')
```
None of this occurred building locally whether in or out of a container. I could, however, pull the production container produced by my CI and reproduce the issue.

After more head scratching than I'd care to admit, I discovered that my CI build does `dotnet publish --no-build`. After comparing the build artifacts, I discovered some files in the wwwroot folder that were different.

.NET 8 SDK with `dotnet publish --no-build`:
	• wwwroot/MyProjectName.modules.json

.NET 9 SDK with `dotnet publish --no-build`:
	• wwwroot/jsmodules.publish.manifest.json

.NET 9 SDK with `dotnet publish` (`--no-build` removed):
	• wwwroot/MyProjectName.modules.json

The contents of the json files are exactly the same.

It appears that something in .NET 9 SDK with regards to the use of `--no-build` has changed copying the files with unexpected names.

## Workaround

Current workaround is to remove the use of the `--no-build` parameter for the `publish` command or rename the files yourself.

### Expected Behavior

The `MyProjectName.modules.json` files should be produced in the publish output as expected.

```shell
dotnet build -c Release
# I typically run `dotnet test -c Release --no-build` here before publishing
dotnet publish --no-build # `dotnet publish` does Release configuration by default these days
```

The commands above should produce the `wwwroot/MyProjectName.modules.json` file rather than `jsmodules.publish.manifest.json`.

Using the .NET 8 SDK, everything worked as expected with the use of `--no-build`.

### Steps To Reproduce

See <https://github.com/craigktreasure/aspnet-58321-repro> for a full repro of the issue.

Build a vanilla .NET 8 Fluent UI Blazor project using the following:

```shell
dotnet build -c Release
# I typically run `dotnet test -c Release --no-build` here before publishing
dotnet publish --no-build # `dotnet publish` does Release configuration by default these days
```

This will produce the `wwwroot/jsmodules.publish.manifest.json` file in the publish output.

Run the application and view the page to observe a visibly blank page.

### .NET Version

9.0.100-rc.1.24452.12 and 9.0.100-rc.2.24474.11

**Comment by craigktreasure** [comment]
@dotnet-policy-service I have added a minimal repro at `https://github.com/craigktreasure/aspnet-58321-repro`.

**Comment by javiercn** [comment]
@craigktreasure thanks for the additional details.

Found the issue. Was likely moved into a target, and doesn't get set if the `--no-build` flag is present. To workaround this issue you can simply set it on your csproj the same way we do (see below).

```
<JSModuleManifestRelativePath Condition="'$(JSModuleManifestRelativePath)' == ''">$(PackageId).modules.json</JSModuleManifestRelativePath>
```

**Comment by javiercn** [comment]
This should take care of it https://github.com/dotnet/sdk/pull/44160

**Comment by javiercn** [comment]
#fixed by https://github.com/dotnet/sdk/p#58269

**Comment by javiercn** [comment]
Fixed by https://github.com/dotnet/sdk/pull/44160

**Comment by javiercn** [comment]
@Bouke are you saying that the fix in https://github.com/dotnet/aspnetcore/issues/58321#issuecomment-2411598004 doesn't work for you?

If so, can you file a separate issue and provide a repro? Note that you can also call publish instead of `publish --no-build`

**Comment by Bouke** [comment]
@javiercn it doesn't work; the published file is named `.modules.json`. I'm now testing this workaround:

```xml
    <PropertyGroup Label=".NET 9 SDK shenanigans">
        <JSModuleManifestRelativePath Condition="'$(JSModuleManifestRelativePath)' == ''">$(AssemblyName).modules.json</JSModuleManifestRelativePath>
    </PropertyGroup>
```

**Comment by craigktreasure** [comment]
Agree. This is not a new issue. The original issue is not fixed. I've updated my [repro](https://github.com/craigktreasure/aspnet-58321-repro) to the .NET 9.0.100 SDK and the issue still exists.

**Comment by craigktreasure** [comment]
@javiercn Can you reopen this issue please? It doesn't appear to have been fixed.

**Comment by alefcarlos** [comment]
Same error here, I have upgraded my net8 to net9.

That does not solve the issue :(

```
<JSModuleManifestRelativePath Condition="'$(JSModuleManifestRelativePath)' == ''">$(PackageId).modules.json</JSModuleManifestRelativePath>
```

![Image](https://github.com/user-attachments/assets/bcc40efb-d1c7-456a-bb70-288af29b4251)

**Comment by alefcarlos** [comment]
@craigktreasure for now I'm using `MSBuildProjectName`:

```xml
<JSModuleManifestRelativePath Condition="'$(JSModuleManifestRelativePath)' == ''">$(MSBuildProjectName).modules.json</JSModuleManifestRelativePath>
```

That fixed the issue.
